core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -42,8 +42,8 @@
 
   // lazily initialized
   private List<ManifestFile> manifests = null;
-  private List<DataFile> adds = null;
-  private List<DataFile> deletes = null;
+  private List<DataFile> cachedAdds = null;
```

Comment:
What rule required this change?

Reply:
There's local vars in `cacheChanges` below that use the names `adds` and `deletes` that can't hide these fields.